weld/jpa/src/main/java/org/jboss/as/weld/services/bootstrap/WeldJpaInjectionServices.java

```diff
@@ -72,7 +80,11 @@ public WeldJpaInjectionServices(DeploymentUnit deploymentUnit) {
         //now we have the service controller, as this method is only called at runtime the service should
         //always be up
         final PersistenceUnitServiceImpl persistenceUnitService = (PersistenceUnitServiceImpl) serviceController.getValue();
-        return new EntityManagerResourceReferenceFactory(scopedPuName, persistenceUnitService.getEntityManagerFactory(), context, deploymentUnit.getAttachment(JpaAttachments.TRANSACTION_SYNCHRONIZATION_REGISTRY), ContextTransactionManager.getInstance());
+        if (persistenceUnitService.getEntityManagerFactory() != null) {
```

Comment:
The enhancement here is to return a LazyEntityManager if we detect that the persistence unit EntityManagerFactory is still null (meaning it hasn't started yet).